### PR TITLE
Add custom simple validations on submitted documents

### PR DESCRIPTION
**Pull request by saalistaja**
- check document type
- check card side

### CODE REVIEW

File: enrollment-server/src/main/java/com/wultra/app/docverify/zenid/provider/ZenidDocumentVerificationProvider.java [hunk 1]
```diff
@@ -499,4 +544,29 @@ private DocumentVerificationStatus toStatus(ZenidWebInvestigateResponse.StateEnu
...
+    private DocumentType toDocumentType(ZenidSharedMineAllResult.DocumentRoleEnum documentRoleEnum) {
+        switch (documentRoleEnum) {
+            case DRV:
+                return DocumentType.DRIVING_LICENSE;
+            case IDC:
+                return DocumentType.ID_CARD;
+            case PAS:
+                return DocumentType.PASSPORT;
+            default:
+                return null;
```
Comment (romanstrobl) [inline enrollment-server/src/main/java/com/wultra/app/docverify/zenid/provider/ZenidDocumentVerificationProvider.java:557]: I am not sure about the `null` value. I would like to suggest to add a new `DocumentType.UNKNOWN` enum value, what do you think?

File: enrollment-server/src/main/java/com/wultra/app/docverify/zenid/provider/ZenidDocumentVerificationProvider.java [hunk 2]
```diff
@@ -499,4 +544,29 @@ private DocumentVerificationStatus toStatus(ZenidWebInvestigateResponse.StateEnu
...
+        }
+    }
+
+    private CardSide toCardSide(ZenidSharedMineAllResult.PageCodeEnum pageCodeEnum) {
+        switch (pageCodeEnum) {
```
Comment (romanstrobl) [inline enrollment-server/src/main/java/com/wultra/app/docverify/zenid/provider/ZenidDocumentVerificationProvider.java:562]: I also suggest to handle the `null` value of `pageCodeEnum` parameter just to be sure.

File: enrollment-server/src/main/resources/application.properties [hunk 3]
```diff
@@ -118,6 +118,7 @@ enrollment-server.presence-check.iproov.serviceHostname=${IPROOV_SERVICE_HOSTNAM
 spring.security.oauth2.client.provider.app.token-uri=http://localhost:6060/oauth/token
 
 # ZenID configuration
+enrollment-server.document-verification.zenid.additionalDocSubmitValidationsEnabled=${ZENID_ADDITIONAL_DOC_SUBMIT_VALIDATIONS_ENABLED:false}
```
Comment (romanstrobl) [inline enrollment-server/src/main/resources/application.properties:121]: I am not sure we need to configure this aspect of the application. If the validations work properly, we should probably just use them. We already have a lot of configuration properties.

**Review by romanstrobl — APPROVED** [review]
Nice addition. I have several suggestions for changes, but feel free to ignore them, these are just minor points.

File: enrollment-server/src/main/resources/application.properties (hunk 3 above)
Comment (saalistaja) [reply enrollment-server/src/main/resources/application.properties:121]: My aim was to pilot the feature slowly, just to be able to turn it off when we would need.
I will still add a TODO for now, we can reconsider removing the configuration later.

File: enrollment-server/src/main/java/com/wultra/app/docverify/zenid/provider/ZenidDocumentVerificationProvider.java (hunk 2 above)
Comment (saalistaja) [reply enrollment-server/src/main/java/com/wultra/app/docverify/zenid/provider/ZenidDocumentVerificationProvider.java:567]: The null check is before calling this method, but I fixed it to be more robust at this place.

File: enrollment-server/src/main/java/com/wultra/app/docverify/zenid/provider/ZenidDocumentVerificationProvider.java (hunk 1 above)
Comment (saalistaja) [reply enrollment-server/src/main/java/com/wultra/app/docverify/zenid/provider/ZenidDocumentVerificationProvider.java]: Fixed

**Pull request merged**